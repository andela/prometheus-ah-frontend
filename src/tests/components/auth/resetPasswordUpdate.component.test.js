import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import mockData from '../../__mocks__/mockData';
import ResetPasswordUpdatePage from '../../../components/auth/ResetPasswordUpdate';

let resetPasswordUpdate;
let location;
let wrapper;

const createMockStore = configureMockStore([thunk]);

describe('Reset Password Update', () => {
  beforeEach(() => {
    const { passwordtoken } = mockData;
    const store = createMockStore({});
    resetPasswordUpdate = jest.fn(() => Promise.resolve());
    location = { search: passwordtoken };
    wrapper = shallow(
      <ResetPasswordUpdatePage.WrappedComponent
        resetPasswordUpdate={resetPasswordUpdate}
        location={location}
        store={store}
      />
    );
  });

  test('should correctly render ResetPasswordUpdatePage ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should set password on input change', () => {
    const { password } = mockData;
    const name = 'password';
    const value = password;
    wrapper
      .find('.form-input')
      .at(0)
      .simulate('change', {
        target: { name, value }
      });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should set password_confirmation on input change', () => {
    const { password } = mockData;
    const name = 'password_confirmation';
    const value = password;
    wrapper
      .find('.form-input')
      .at(1)
      .simulate('change', {
        target: { name, value }
      });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should return error for invalid form submission', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state('error')).not.toBe({});
    expect(wrapper).toMatchSnapshot();
  });

  test('should call resetPasswordUpdate props for valid form submission', () => {
    const { password, passwordtoken } = mockData;
    const data = {
      password,
      password_confirmation: password,
      passwordtoken
    };

    const token = passwordtoken.split('=')[1];
    wrapper.setState(data);
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(resetPasswordUpdate).toHaveBeenCalledWith({
      password,
      password_confirmation: password,
      passwordtoken: token
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should call the dispatch mapped to the props', () => {
    const store = createMockStore({});
    const wrapper2 = shallow(
      <ResetPasswordUpdatePage store={store} location={location} />
    );
    expect(wrapper2).toMatchSnapshot();
  });
});
