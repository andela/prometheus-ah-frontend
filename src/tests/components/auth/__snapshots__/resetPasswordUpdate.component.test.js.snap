// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Update should call resetPasswordUpdate props for valid form submission 1`] = `
<div
  className="container py-5 form-gap form-margin"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 mx-auto"
        >
          <div
            className="card rounded-0"
          >
            <div
              className="card-header"
            >
              <h4
                className="modal-title"
              >
                Enter Password
              </h4>
            </div>
            <div
              className="card-body"
            >
              <form
                autoComplete="off"
                className="form mt-2"
                id="formLogin"
                method="post"
                onSubmit={[Function]}
              >
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    field="password"
                    onChange={[Function]}
                    placeholder="enter your password"
                    type="password"
                    value="password"
                  />
                </div>
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    field="password_confirmation"
                    onChange={[Function]}
                    placeholder="confirm your password"
                    type="password"
                    value="password"
                  />
                </div>
                <button
                  className="btn btn-block btn-update login-btn"
                  id="btnLogin"
                  type="submit"
                >
                  Reset Password
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Reset Password Update should call the dispatch mapped to the props 1`] = `
<ResetPasswordUpdate
  location={
    Object {
      "search": "?emailToken=faketoken",
    }
  }
  resetPasswordUpdate={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`Reset Password Update should correctly render ResetPasswordUpdatePage  1`] = `
<div
  className="container py-5 form-gap form-margin"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 mx-auto"
        >
          <div
            className="card rounded-0"
          >
            <div
              className="card-header"
            >
              <h4
                className="modal-title"
              >
                Enter Password
              </h4>
            </div>
            <div
              className="card-body"
            >
              <form
                autoComplete="off"
                className="form mt-2"
                id="formLogin"
                method="post"
                onSubmit={[Function]}
              >
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    field="password"
                    onChange={[Function]}
                    placeholder="enter your password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    field="password_confirmation"
                    onChange={[Function]}
                    placeholder="confirm your password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-block btn-update login-btn"
                  id="btnLogin"
                  type="submit"
                >
                  Reset Password
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Reset Password Update should return error for invalid form submission 1`] = `
<div
  className="container py-5 form-gap form-margin"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 mx-auto"
        >
          <div
            className="card rounded-0"
          >
            <div
              className="card-header"
            >
              <h4
                className="modal-title"
              >
                Enter Password
              </h4>
            </div>
            <div
              className="card-body"
            >
              <form
                autoComplete="off"
                className="form mt-2"
                id="formLogin"
                method="post"
                onSubmit={[Function]}
              >
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    error={
                      Array [
                        "password field is required.",
                      ]
                    }
                    field="password"
                    onChange={[Function]}
                    placeholder="enter your password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group password-input"
                >
                  <TextField
                    className="form-control form-input"
                    error={
                      Array [
                        "The password confirmation field is required.",
                      ]
                    }
                    field="password_confirmation"
                    onChange={[Function]}
                    placeholder="confirm your password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-block btn-update login-btn"
                  id="btnLogin"
                  type="submit"
                >
                  Reset Password
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
